upload.go

```diff
@@ -125,6 +144,118 @@ type s3repo struct {
 	session    *session.Session
 }
 
+func NewB2Repo(opts options) (*b2repo, error) {
+	r := &b2repo{
+		appKey:            opts.B2AppKey,
+		bucket:            opts.B2Bucket,
+		concurrentUploads: opts.B2ConcurrentUploads,
+		ctx:               context.Background(),
+		endpoint:          opts.B2Endpoint,
```

Comment:
`endpoint` is not used but it seems possible to give a custom endpoint with `b2.APIBase()`

Reply:
Hi, I tried to find out if this is useful for anything and did not find why we need this. It looks like it is only necessary if we want to create S3 compatible URLs, where I see no use case for pg_back. I searched a little bit and all B2 compatible tools only use `keyID` and `appKey`. I think the region (US or EU) is baked into the keyID or bucket.